spec/datadog/appsec/waf_spec.rb

```diff
@@ -1664,21 +1664,42 @@
           Datadog::AppSec::WAF::Handle.new(rule, limits: { max_container_depth: 1 })
         end
 
-        let(:matching_input) do
-          { 'server.request.headers.no_cookies' => { 'user-agent' => ['Nessus SOAP'] } }
+        context 'when matching input is outside the limit' do
+          let(:matching_input) do
+            { 'server.request.headers.no_cookies' => { 'user-agent' => ['Nessus SOAP'] } }
+          end
+
+          it 'passes' do
```

Comment:
```
it 'passes on matching input outside of limit'
```

Reply:
I believe that is somewhat redundant since the context block from above provides the `input outside of limit` context.

![image](https://user-images.githubusercontent.com/4672858/217565083-bd4f1c52-9705-4aad-9904-64388a390043.png)

Reply:
I'm ok changing it, but I believe the Rspec framework has all these concepts of context, describe to build a readable out put. 

What do you think @lloeki ?

Reply:
When reviewing the code I felt it had the potential for ambiguity (in fact it made me pause for a moment, reread the strings and the body back and forth for a few seconds until it clicked).

The initial code was something like:

```
context 'when matching input is outside of limit' do
  it 'passes` do
```

Which seemed to me as if `context` corresponded to the expected test case result (we can't be matching the input if we're not running anything).

So I suggested:

- `context 'when value is outside of limit' do` is about the surrounding context in which we will execute the test, hence the wording using `value`
- `it 'passes on matching input outside of limit'` is about the expected behaviour to test against execution.

This comes into play as we may add or factor in more tests to a given context or nest another context within, and possibly change the context itself. The meaning of a freestanding `it 'passes'` is then lost.

Reply:
Thanks for the nice explanation. I agree with the sentiment of `it 'passes'` might not have all the necessary details of the actual spec case, and if in the future we add more tests to the matrix we might create some confusion. 

Changed to the suggested message.